lib/puppet/application/agent.rb

```diff
@@ -71,6 +71,10 @@ def preinit
     options[:waitforcert] = arg.to_i
   end
 
+  option("--job-id ID") do |arg|
+    options[:job_id] = arg
```

Comment:
just to confirm - this is OK as any kind of string, ie UTF-8 will make through the system ok?